radixdlt-java/src/main/java/com/radixdlt/client/application/translate/atomic/AtomicToParticlesMapper.java

```diff
@@ -0,0 +1,24 @@
+package com.radixdlt.client.application.translate.atomic;
+
+import com.radixdlt.client.application.translate.Action;
+import com.radixdlt.client.application.translate.StatelessActionToParticlesMapper;
+import com.radixdlt.client.core.atoms.particles.SpunParticle;
+import io.reactivex.Observable;
+
+public class AtomicToParticlesMapper implements StatelessActionToParticlesMapper {
+	@Override
+	public Observable<Action> sideEffects(Action action) {
+		if (!(action instanceof AtomicAction)) {
```

Comment:
Is there a reasonable case where this might be called with something that is not an AtomicAction? Could we make StatelessActionToParticlesMapper generic for type safety (e.g. `StatelessActionToParticlesMapper<T extends Action>`  and `sideEffects(T action)`)

Reply:
Yes, a lot of usecases as described above. Another arbitrary example, you can look at every action and create a MessageAction log sent to a different address (of course you need to worry about infinite recursion in this case).